lib/ts/types.ts

```diff
@@ -30,7 +30,13 @@ export type GetRedirectionURLContext = {
 };
 
 export type ValidationFailureCallback =
-    | (({ userContext, reason }: { userContext: any; reason: any }) => Promise<string | undefined> | string | undefined)
+    | (({
+          userContext,
+          reason,
+      }: {
+          userContext: any;
+          reason: any;
+      }) => Promise<string | undefined | null> | string | undefined | null)
```

Comment:
In this context does a `null` return value have a separate meaning from `undefined`? 
I'd prefer if this either:
- stayed as `string | undefined` or
- if this was handled identically to `undefined`. (In the above implementation a `null` return value disables the access denied screen, but I don't think that's intentional)

Reply:
We had a discussion regarding this and the concern was that 

```
onFailureRedirct: () => undefined, showAccessDeniedOnFailure: false
```
would be same as 

`onFailureRedirct: () => null`

To avoid confusion, we didn't wanna have the option to return `null` as well.


However, this is not true. In the first case, the [default redirection](https://github.com/supertokens/supertokens-auth-react/blob/master/lib/ts/recipe/session/recipe.tsx/#L196-L198)  ('/') would still happen while in the latter it won't. We still need to support null to make use cases like `/auth/verify-phone` after login work.